manifests/snmpv3_user.pp

```diff
@@ -54,6 +59,7 @@
 #
 define snmp::snmpv3_user (
   $authpass,
+  $engineid,
```

Comment:
Your docs say $engineid is optional and then you require a value.  I suggest `$engineid = unset`.